### PR TITLE
Reef 336

### PR DESCRIPTION
**Pull request by beysims**

### CODE REVIEW

**Comment by markusweimer** [comment]
Please merge the PR into a single commit.

File: lang/cs/Org.Apache.REEF.Driver/Defaults/DefaultDriverStartHandler.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Time.Event;
+
+namespace Org.Apache.REEF.Driver.Defaults
+{
+    /// <summary>
+    ///  Default event handler for driver start: Logging it.
+    /// </summary>
+    public class DefaultDriverStartHandler : IObserver<DateTime>
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Driver/Defaults/DefaultDriverStartHandler.cs]: Having a default for start seems dangerous. What if the user doesn't bind a start handler? Would we immediately shutdown or create an orphaned driver? The latter would be very bad, but even the former seems like a waste of cycles, as that driver wouldn't do anything.

That's why on the java side, this is the one event the application must provide an event handler for.

Comment (beysims) [inline lang/cs/Org.Apache.REEF.Driver/Defaults/DefaultDriverStartHandler.cs]: Agreed but this would be breaking change. Is this acceptable?

Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Driver/Defaults/DefaultDriverStartHandler.cs]: Ah, I get it. We need to remove this class once 0.12 comes around. Can you file a JIRA to remind us?

**Pull request closed without merging**